server/src/streaming/partitions/storage.rs

```diff
@@ -221,7 +221,7 @@ impl Storage<Partition> for FilePartitionStorage {
             );
             segment.load().await?;
             if !segment.is_closed {
-                segment.unsaved_messages = Some(Vec::new())
+                segment.unsaved_batches = Some(Vec::new())
```

Comment:
consider using `Vc` from `atone` instead of `Vec`

Reply:
Going to do that in another pull request, proper struct wrapping unsaved_batches buffer is required in order to expose api, similar to Vec<T>

Reply:
ok